### ISSUE TITLE
Enough exp to level up, enough damage to get killed

### ISSUE THREAD

**Issue by robertoschiavone** [HabitRPG/habitica#8462]
[//]: # (Before logging this issue, look through common problems at https://github.com/HabitRPG/habitrpg/issues If you find your issue there, read at least the first post to see if there is a workaround for you)

[//]: # (Github is primarily used for reporting bugs. If you have a feature request, use "Help > Request a Feature" so that the feature request can be vetted by the larger Habitica community)

[//]: # (To report a bug in one of the mobile apps, please report it in the correct repository. Android: https://github.com/HabitRPG/habitrpg-android, iOS: https://github.com/HabitRPG/habitrpg-ios)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitrpg/issues/2760)

[//]: # (Fill out relevant information - UUID is found in Settings -> API)
General Info
  * UUID: a05dda3d-3343-4633-8853-05426ffa951b
  * Browser: Mozilla Firefox 51.0.1 
  * OS: macOS Sierra 10.12.2

### Description
[//]: # (Describe bug in detail here. Include pictures if helpful.)
Quest: Vice Part 3: Vice Awakens (1000 exp, 100 gold, other stuff)
My level before finishing the quest: 52
When the boss was killed, I got enough experience to level up to level 53, but also enough damage to get killed. As result, I lost all my gold and my level dropped to 51 (0 exp).
I think the outcome should have been one of the following two:
- I get all the rewards and level up to level 53, then die and go back to level 52;
- I die and go back to level 51, then get the rewards and end up with 1000 exp and 100 gold.

#### Console Errors
[//]: # (Include any JavaScript console errors here.)

**Comment by lemoness** [comment]
The correct order of operations should be:

1) The player dies, loses a level, and loses all their current gold.
2) The player receives the benefits for the quest, including XP and gold, and gets to keep those going forward.

**Comment by Germlord** [comment]
This is pretty old, but I'll see if I can repro this/fix it if it's still there.

**Comment by Germlord** [comment]
My first attempt to reproduce this with debug resulted in killing the boss, but not inflicting any damage for missing dailies at all--it appears that part of Cron might have been skipped? It's worth mentioning that this was after I manually set Cron, on the same day that the account was created. Unfortunately, I won't have time to further investigate, so I would encourage any other volunteers to go ahead and make their own attempts at this bug.

tl;dr: Bug couldn't be reproduced, might have been masked by other, related bugs. Will be leaving this issue to the next volunteer.

**Comment by Alys** [comment]
@Germlord Was the Start Date of the Dailies set to be in the past? If it wasn't, then cron would have correctly ignored them.
I.e., if you created a test Daily today, it's default start date would have been today. If you did not change the Start Date to yesterday or earlier, then when you caused cron to run today, cron would have seen that the test Daily was not due yesterday, and so it would have ignored it.

**Comment by lilithi** [comment]
I'll try fixing this

**Comment by Alys** [comment]
@lilithi Thanks! Please go ahead and post here if you have questions.

**Comment by shanaqui** [comment]
@lilithi Hi there! Are you still working on this one? If not, or if you don't reply within a week, we're going to mark this as "help wanted" again to make sure people know it's available to work on. Thanks!

**Comment by lilithi** [comment]
Go ahead and change it to help wanted. Unfortunately I've not had the time to mess with it

**Comment by Bokoness** [comment]
Hello, still need help with this issue?

**Comment by anishlukk123** [comment]
Hello, still need help with this issue? @SabreCat

**Comment by Moniarchy** [comment]
Hey! Can I take this on or is the behavior fixed/no longer relevant? I've started trying to repro it and wanted to check first. Would be my second contribution.

**Comment by RishmitaR** [comment]
If this issue still needs to be worked on I would like to work on it! @SabreCat
